lib/seattleflu/db/cli/command/longitudinal.py

```diff
@@ -0,0 +1,355 @@
+    "stub" swabname if the original stub were not converted to a longer or more
+    unique stubname like "swab_collected_x".
+    """
+    rename_map = {}
+
+    swab_columns = list(df.filter(regex=regex, axis=1))
+    for col in swab_columns:
+        rename_map[col] = re.sub(current_stub, desired_stub, col)
+
+    return df.rename(columns=rename_map)
+
+
+def longitudinal_stubnames(df: pd.DataFrame) -> [str]:
+    """
+    Returns a list of the stubnames of longitudinal columns that have suffixes
+    depicting the week number of the study (e.g. 'swab_date_11').
+
+    TODO: better optimized way to apply this, e.g. via `filter`?
+    """
+    stubnames = []
+    for column in list(df):
+        match = re.match(".*(?=_[0-9]+)", column)
+        if match:
+            stubnames.append(match[0])
+
+    return list(set(stubnames))
+
+
+def select_all_that_apply_wide_to_long(df: pd.DataFrame) -> pd.DataFrame:
+    """
+    Given a wide DataFrame *df*, collapses "Select All That Apply"-type
+    categorical survey responses with pseudo-one-hot-encoding format and
+    converts the responses into human-readable lists of equivalent logical
+    value.
+
+    Throws a :class:`AssertionError` if the number of rows in the resulting
+    DataFrame differs from the given *df*.
+
+    Contains hard-coded column names that may need to be updated in the future.
+    """
+    df = rename_short_stub(df, regex='household_(?!children|ill)',
+                           current_stub='household', desired_stub='household_sx')
+
+    stubnames = [
+        "race", "feeding", "meds", "rn_sx", "sx_specific", "missed_impact",
+        "household_sx", "travel_type", "household_ill",
+    ]
+    pid = ['study_id'] + ['week']
+
+    stubbed_columns = list(df.filter(regex='|'.join(stubnames)))
+    reshaped_data = df.drop(stubbed_columns, axis=1)
+
+    for stub in stubnames:
+        true_values = collapse_wide_stubbed_columns(df, stub, pid)
+        if true_values is None:
+            continue
+
+        reshaped_data = reshaped_data.merge(true_values, how='left', on=pid)
+
+    assert len(df) == len(reshaped_data), f"You do not have a 1:1 merge on {pid}"
+
+    return reshaped_data
+
+
+def collapse_wide_stubbed_columns(df: pd.DataFrame, stub: str,
+                                  pid: str) -> pd.DataFrame:
+    """
+    Takes a wide DataFrame *df* with columns partially matching the given
+    *stub*. For each row, collapses the values from the matching column-group
+    with the same *pid* into a list containing structurally different but
+    logically equivalent values as the original DataFrame.
+    """
+    stubset = df.filter(regex=f"{'|'.join(pid)}|{stub}_")
+    if len(list(stubset)) == len(pid):
+        LOG.warning(f"Stub {stub} not found in data")
+        return None
+
+    long_stubset = pd.wide_to_long(stubset, stub, i=pid, j=f"new_{stub}",
+                                   sep="_", suffix='\\w+'
+                                  ).reset_index()
```

Comment:
I don't understand why this `wide_to_long` fails without the provided `suffix` argument. The same function works fine without `suffix` when casting longitudinal columns from `wide_to_long`. I feel like adding a "it works don't touch" inline comment here :grimacing: